rotkehlchen/exchanges/coinbase.py

```diff
@@ -160,20 +167,108 @@ def __init__(
             secret=secret,
             database=database,
         )
+
+        self.is_legacy_api_key = self.is_legacy_key(api_key)
```

Comment:
As I wrote here: https://github.com/rotki/rotki/pull/7903#discussion_r1593532832 do not call this here.
Read that comment to understand why.
```suggestion

```

Reply:
I've incorporated the method call within validate_api_key. However, I'm encountering unexpected behavior during local testing. It appears that validate_api_key does not execute when I restart the app locally with credentials already saved. Editing or re-adding the keys resolves the issue, and logging out and back in also bypasses the problem. However, declaring it at the instance level seems to solve this. Is this behavior expected?

Reply:
Hmm @vijithreddy  okay then this only runs when the exchange is edited or setup. Then add it here too **but* make sure to add it under try catch for the errors the function adds. And log an error if it happens and set it to legacy.